Meta/CMake/Superbuild/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+set(LAGOM_ONLY_VARS "CMAKE_CXX_COMPILER;CMAKE_C_COMPILER")
+set(EXCLUDE_FROM_CL_VARS "SERENITY_ARCH;SERENITY_TOOLCHAIN")
+
+# https://stackoverflow.com/a/48555098
+# https://cmake.org/pipermail/cmake/2018-January/067002.html
+# MUST be done before call to 'project'
+get_cmake_property(vars CACHE_VARIABLES)
+foreach(var IN LISTS vars)
+    get_property(currentHelpString CACHE "${var}" PROPERTY HELPSTRING)
+    if("${currentHelpString}" MATCHES "No help, variable specified on the command line." OR "${currentHelpString}" STREQUAL "")
+        message("${var} = [${${var}}]  --  ${currentHelpString}") # uncomment to see the variables being processed
+        if (var IN_LIST EXCLUDE_FROM_CL_VARS)
+            continue()
+        elseif(var IN_LIST LAGOM_ONLY_VARS)
+            list(APPEND LAGOM_CL_ARGS "-D${var}=${${var}}")
+        else()
+            list(APPEND CL_ARGS "-D${var}=${${var}}")
+        endif()
+    endif()
+endforeach()
+
+list(APPEND LAGOM_CL_ARGS ${CL_ARGS})
+
+project(
+    SerenitySuperbuild
+    DESCRIPTION "Orchestrate host and target builds in a single build"
+    LANGUAGES NONE
+)
+
+get_filename_component(
+    SERENITY_SOURCE_DIR "${PROJECT_SOURCE_DIR}/../../.."
+    ABSOLUTE CACHE
+)
+set(SERENITY_ARCH "i686" CACHE STRING "Architecture to build for (i686 or x86_64)")
+set(SERENITY_TOOLCHAIN "GNU" CACHE STRING "Compliler toolchain to use for Serenity (GNU or Clang)")
+set(SERENITY_BUILD_DIR "${PROJECT_BINARY_DIR}/../${SERENITY_ARCH}")
+
+include(ExternalProject)
+
+ExternalProject_Add(
+    lagom
+    SOURCE_DIR "${SERENITY_SOURCE_DIR}/Meta/Lagom"
+    BINARY_DIR "${PROJECT_BINARY_DIR}/../lagom"
+    INSTALL_DIR "${PROJECT_BINARY_DIR}/../lagom-install"
+    CMAKE_GENERATOR Ninja
+    EXCLUDE_FROM_ALL YES
+    CMAKE_ARGS ${LAGOM_CL_ARGS}
+    CMAKE_CACHE_DEFAULT_ARGS
+    "-DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>"
+    "-DUNICODE_CACHE_LOCATION:PATH=${SERENITY_BUILD_DIR}"
+    "-DBUILD_LAGOM:BOOL=${BUILD_LAGOM}"
+    # Always call the build step of tools, so keeping things up-to-date is easy
+    BUILD_ALWAYS YES
+    # Expose install step as a target, so it can be depended on
+    STEP_TARGETS install
+)
+
+ExternalProject_Add(
+    serenity
+    SOURCE_DIR "${SERENITY_SOURCE_DIR}"
+    BINARY_DIR "${SERENITY_BUILD_DIR}"
+    CMAKE_GENERATOR Ninja
+    CMAKE_ARGS ${CL_ARGS}
+    CMAKE_CACHE_DEFAULT_ARGS
+    # Tell the find_package(Lagom REQUIRED) command call where to find
+    # the CMake package
+    "-DCMAKE_PREFIX_PATH:PATH=${PROJECT_BINARY_DIR}/../lagom-install"
+    "-DUNICODE_CACHE_LOCATION:PATH=${SERENITY_BUILD_DIR}"
+    "-DSERENITY_ARCH:STRING=${SERENITY_ARCH}"
+    "-DSERENITY_TOOLCHAIN:STRING=${SERENITY_TOOLCHAIN}"
+    # Always call the build step
+    BUILD_ALWAYS YES
+    # Host tools must be built and installed before the OS can be built
+    DEPENDS lagom-install
+    STEP_TARGETS install
+)
```

Comment:
Putting this at then end.  
Please consider adding https://github.com/friendlyanon/cmake-init-gif-engine/commit/662ca8894d833c8c945c537c5d97d95dc6ffa4cf to this as well. See the commit message for reasoning.

Reply:
Seems reasonable. I think the Makefile generator is currently broken for reasons related to the LibC build (go figure :D). But there's no sense having known bugs impact that and future(?) Visual Studio generator support.